cmd/ack-generate/command/crossplane.go

```diff
@@ -67,24 +68,29 @@ func generateCrossplane(_ *cobra.Command, args []string) error {
 			return fmt.Errorf("cannot get the API model for service %s", svcAlias)
 		}
 	}
-	cfgPath := filepath.Join(providerDir, "apis", svcAlias, optGenVersion, "generator-config.yaml")
+	cfgPath := ""
+	gcPath := filepath.Join(providerDir, "apis", svcAlias, optGenVersion, "generator-config.yaml")
+	if _, err := os.Stat(gcPath); !os.IsNotExist(err) {
+		cfgPath = gcPath
+	}
 	g, err := generate.New(
-		sdkAPI, optGenVersion, cfgPath, cpgenerate.DefaultConfig,
+		sdkAPI, optGenVersion, cfgPath, crossplane.DefaultConfig,
 	)
 	if err != nil {
 		return err
 	}
-
-	ts, err := cpgenerate.Crossplane(g, optTemplatesDir)
+	ts := templateset.New(optTemplatesDir, crossplane.IncludePaths, crossplane.CopyPaths, crossplane.TemplateFuncs)
+	generation := crossplane.NewGeneration(g, ts,
+		crossplane.WithInitializer(crossplane.AddAPIFiles),
+		crossplane.WithInitializer(crossplane.AddCRDFiles),
+		crossplane.WithInitializer(crossplane.AddControllerFiles),
+	)
+	output, err := generation.Run()
```

Comment:
I'm a little confused why you went back to the Generation.Run() thing here... what wasn't possible to do with the `pkg/generate/crossplane.Crossplane` function?

Reply:
The fixes `pkg/generate/crossplane.Crossplane` needed weren't that big but I'm planning to write the tests separately for each one of those initializers instead of testing the whole pipeline. Also wanted to separate the core logic in `Run` so that when we a developer adds a new initializer, they don't have to understand the whole flow.

Reply:
I'm still not really liking the whole "Initializer" thing here... I was trying to encapsulate all the necessary logic regarding building the templates into the `pkg/generate/crossplane`  package instead of having to have the CLI tool contain this logic.

Reply:
@muvaf OK, I pushed up an alternative fix for this. Please let me know whether you'd be OK with it.

https://github.com/aws/aws-controllers-k8s/pull/593

Reply:
Sounds good to me. I've just tested your [PR](https://github.com/aws/aws-controllers-k8s/pull/593) and I think we can close this one.